lib/controller.js

```diff
@@ -150,15 +166,22 @@ Tessel.get = function(opts) {
     function logAndFinish(tessel) {
       // The Tessels that we won't be using should have their connections closed
       var connectionsToClose = tessels;
-
       if (tessel) {
-        logs.info(sprintf('Connected to %s over %s', tessel.name, tessel.connection.connectionType));
-        connectionsToClose.splice(tessels.indexOf(tessel), 1);
+        if (opts.root) {
 
-        controller.closeTesselConnections(connectionsToClose)
-          .then(function() {
-            return resolve(tessel);
-          });
+          controller.closeTesselConnections(connectionsToClose)
+            .then(function() {
+              controller.ssh.runSSH(0, opts, tessels, resolve, reject);
```

Comment:
@Student007, you shouldn't need to modify the `Tessel.get` function. It should be a black box that returns Tessels to functions that ask for it and shouldn't have any references to any other specific Tessel commands. You should call the `ssh` function from within `controller.root` after requesting the Tessel from `Tessel.get`.